llvm/include/llvm/MC/MCInstrItineraries.h

```diff
@@ -165,8 +186,8 @@ class InstrItineraryData {
 
   /// Return the cycle for the given class and operand. Return std::nullopt if
   /// the information is not available for the operand.
-  std::optional<unsigned> getOperandCycle(unsigned ItinClassIndx,
-                                          unsigned OperandIdx) const {
+  virtual std::optional<unsigned> getOperandCycle(unsigned ItinClassIndx,
+                                                  unsigned OperandIdx) const {
```

Comment:
It would be easiest if it would look like itineraries...
Also, I have seen updates in itineraries upstream, like have an source-level enumeration over all itineraries.